Hearthstone Deck Tracker/Windows/OverlayWindow.xaml.cs

```diff
@@ -370,7 +371,20 @@ internal async Task ExperienceChangedAsync(int experience, int experienceNeeded,
 			}
 			if(_game.CurrentMode != Enums.Hearthstone.Mode.HUB)
 				HideExperienceCounter();
-
+		}
+		internal void UpdateOpponentDeadForTurns(List<int> turns)
+		{
+			var index = _leaderboardDeadForText.Count - 1;
+			foreach(var text in _leaderboardDeadForText)
+				text.Text = "";
+			foreach(var text in _leaderboardDeadForTurnText)
+				text.Text = "";
+			foreach(var turn in turns)
+			{
+				_leaderboardDeadForText[index].Text = $"{turn}";
+				_leaderboardDeadForTurnText[index].Text = "turns";
```

Comment:
Using an index that is not based on the size of the array is a bit dangerous. You should verify that the index is in range of both of the arrays.